app/controllers/admin/materials_controller.rb

```diff
@@ -27,20 +28,36 @@ def index
     end
 
     def create
-      @material = Material.new(set_params)
-      if @material.save
+      if @material.save && @valid
         flash[:notice] = 'Material created successfully!'
         render js: "window.location='#{admin_materials_path}'"
       else
         flash.now[:error] = @material.errors.full_messages
+        add_error
         render_add
       end
     end
 
     private
 
     def set_params
-      params.require(:material).permit(:name, :material_type, :status)
+      params.require(:material).permit(:name, :material_type, :status, :serial_number)
+    end
+
+    def create_params
+      @material = Material.new(set_params)
+      if @material.material_type == 'License'
+        @valid = @material.serial_number.match?(/\d{3}\-\d{3}\-\d{3}/)
```

Comment:
This logic (checking for serial number format) is meant more for the model side of code instead of the controller. If you add this format checking as a validation for `Material` model, you won't have to change the `create` method here or need to use the `create_params` method.